plc4j/api/src/main/java/org/apache/plc4x/java/api/PlcDriverManager.java

```diff
@@ -16,17 +16,25 @@
  * specific language governing permissions and limitations
  * under the License.
  */
-package org.apache.plc4x.java.utils.connectionpool2;
+package org.apache.plc4x.java.api;
 
-import org.apache.plc4x.java.api.PlcConnection;
+import org.apache.plc4x.java.DefaultPlcDriverManager;
 import org.apache.plc4x.java.api.exceptions.PlcConnectionException;
 
-/**
- * Simple Factory Method which creates a new PlcConnection
- */
-@FunctionalInterface
-public interface PlcConnectionFactory {
+import java.util.Set;
+
+public interface PlcDriverManager {
+
+    static PlcDriverManager getDefault() {
+        return new DefaultPlcDriverManager();
+    }
+
+    Set<String> listDrivers();
+
+    PlcDriver getDriver(String protocolCode) throws PlcConnectionException;
+
+    PlcDriver getDriverForUrl(String url) throws PlcConnectionException;
 
-    PlcConnection create() throws PlcConnectionException;
```

Comment:
I wonder if we should leave the create method as deprecated default implementation:
```
default PlcConnection create() {
    return getConnectionManager().create();
}
```

Reply:
What for? I mean ... we're obviously not using it.

Reply:
because it is part of the API package an might break exiting deployments from customers.